### PR TITLE
feat: double opt-in registration flow

### PR DESCRIPTION
**Pull request by mkucmus**
## Changes

closes: #1353

### Checklist

- [x] the [list of features](docs/guide/FEATURELIST.md) is updated (if it's a feature and it's relevant)
- [x] I followed [contributing](https://github.com/DivanteLtd/shopware-pwa/blob/master/CONTRIBUTING.md) guidelines

### CODE REVIEW

File: packages/default-theme/src/pages/registration/confirm.vue [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+      () => success.value,
+      () => {
+        if (success.value) {
+          refreshUser()
+          redirectTo && redirectUser(redirectTo)
+        }
+      }
+    )
+
+    onDeactivated(() => clearTimeout())
```
Comment (mkucmus) [inline packages/default-theme/src/pages/registration/confirm.vue:110]: TODO ;)

File: packages/default-theme/src/pages/checkout.vue [hunk 2]
```diff
@@ -164,7 +164,13 @@ export default {
       if (!isFormCorrect) {
         return
       }
-      await register(registrationFormData.value)
+
+      // register user and point /checkout path that may be used in double opt-in (optional usage)
+      await register(
+        Object.assign({}, registrationFormData.value, {
+          redirectTo: encodeURIComponent(PAGE_CHECKOUT),
```
Comment (patzick) [inline packages/default-theme/src/pages/checkout.vue:171]: we should use this for url, to save domain settings
```suggestion
          redirectTo: routing.getUrl(PAGE_CHECKOUT),
```

`routing` object can be fetched from `getApplicationContext`

File: packages/default-theme/src/pages/checkout.vue [hunk 3]
```diff
@@ -208,6 +214,7 @@ export default {
         // perform a redirection to the external payment gateway
         window.location.href = redirectUrl
       } catch (error) {
+        console.warn("err place order", error)
```
Comment (patzick) [inline packages/default-theme/src/pages/checkout.vue:217]: `const {devtools} = getApplicationContext()`
```suggestion
        devtools?.warning("[checkout] Error while placing an order", error)
```

File: packages/default-theme/src/pages/registration/confirm.vue [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+      },
+    } = root
+
+    const success = ref(false)
+    const errors = ref([])
+    const loading = ref(true)
+    const redirectTimer = ref()
+    const redirectUser = (redirectTo) =>
+      setTimeout(() => {
+        router.push(redirectTo)
```
Comment (patzick) [inline packages/default-theme/src/pages/registration/confirm.vue:80]: is the domain correct here?

**Review by patzick — CHANGES_REQUESTED** [review]
Few suggestions from me

File: packages/default-theme/src/pages/registration/confirm.vue (hunk 4 above)
Comment (mkucmus) [reply packages/default-theme/src/pages/registration/confirm.vue:80]: we pass here only a path, which is being used in `router.push` once user is confirmed. It's definitely case to be checked 👍🏻

File: packages/default-theme/src/pages/checkout.vue (hunk 2 above)
Comment (mkucmus) [reply packages/default-theme/src/pages/checkout.vue:171]: thanks, hawk-eye! without routing.getUrl we are always redirected to the default language without any prefix 🙏🏻 now it's fixed. I left encodeURIComponent, path is automatically decoded on vue side 👍🏻

File: packages/default-theme/src/pages/checkout.vue (hunk 3 above)
Comment (mkucmus) [reply packages/default-theme/src/pages/checkout.vue:217]: done

File: packages/default-theme/src/pages/registration/confirm.vue (hunk 4 above)
Comment (mkucmus) [reply packages/default-theme/src/pages/registration/confirm.vue:80]: I checked it and router pushes the previously configured path (prepended with current domain' path) as it's passed during registration process. so it's working great!

**Review by patzick — APPROVED** [review]
®️ ✅

**Pull request merged**